commands.js

```diff
@@ -813,13 +815,22 @@ exports.commands = {
 			room.auth[userid] = nextGroup;
 		}
 
+		// Only show popup if: user is online and in the room, the room is public, and not a groupchat or a battle.
+		let needsPopup = targetUser && Rooms.rooms[room.id].users[targetUser.userid] && !room.isPrivate && !room.isPersonal && !room.battle;
```

Comment:
`Rooms.rooms[room.id].users` can be replaced with `room.users`.

@TheImmortal had a brainfart when he wrote that code, but we might as well fix it while we're refactoring here.